OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/AnalyticsUploadHelper.java

```diff
@@ -14,20 +14,21 @@
 import java.io.IOException;
 import java.util.ArrayList;
 
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
 import okhttp3.Response;
 
 class AnalyticsUploadHelper {
+    private static final String TAG = AnalyticsUploadHelper.class.getName();
 
     enum Result {
         SUCCESS,
-        FAILED_RETRY_LATER
-    };
+        FAILED_RETRY_LATER,
+        FAILED_NO_RETRY
```

Comment:
What is failed no retry for / the previous changes in the upload tasks fall through to scheduling for this case, is it correct?

Reply:
(ok see what it's for below, other Q remains)

Reply:
if there are no credentials we wont schedule a retry as we know every retry will fail. Otherwise it would keep retrying with exponential delay. We cannot rely on this as by the time credentials are given a lot of time could theoretically pass

So, a small optimisation (without which could live as well)

Reply:
Yep, makes sense.